File: main-workshop/services/dotnet/dynamic-pricing-service/Models/PricingResult.cs
using System.Text.Json.Serialization;

public class PricingResult
{
    [JsonPropertyName("rideId")]
    public string RideId { get; set; } = string.Empty;
    
    [JsonPropertyName("riderId")]
    public string RiderId { get; set; } = string.Empty;
    
    [JsonPropertyName("finalPrice")]
    public decimal FinalPrice { get; set; }
    
    [JsonPropertyName("basePrice")]
    public decimal BasePrice { get; set; }
    
    [JsonPropertyName("surgeMultiplier")]
    public decimal SurgeMultiplier { get; set; }
    
    [JsonPropertyName("success")]
    public bool Success { get; set; }
    
    [JsonPropertyName("errorMessage")]
    public string ErrorMessage { get; set; } = string.Empty;
    
    [JsonPropertyName("errorType")]
    public string ErrorType { get; set; } = string.Empty;
}

File: main-workshop/services/java/dynamic-pricing-service/src/main/java/com/powertoolsride/dynamicpricingservice/model/PricingResult.java
package com.powertoolsride.dynamicpricingservice.model;

import java.math.BigDecimal;
import com.fasterxml.jackson.annotation.JsonProperty;

public record PricingResult(
    @JsonProperty("rideId") String rideId,
    @JsonProperty("riderId") String riderId,
    @JsonProperty("finalPrice") BigDecimal finalPrice,
    @JsonProperty("basePrice") BigDecimal basePrice,
    @JsonProperty("surgeMultiplier") BigDecimal surgeMultiplier,
    @JsonProperty("success") boolean success,
    @JsonProperty("errorMessage") String errorMessage
) {
}
